development/docker-entrypoint.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -ex
+
+if [[ ! -f index.php ]]; then
+    echo "Error: No source code in /var/www/html – you must mount your code base to that path!"
+    exit 1
+fi
+
+if  [[ "$1" != apache2* && "$1" != php-fpm && "$1" != bin* ]]; then
+    exec $@
+fi
+
+# Ensure two very essential requirements are set in the config file.
+if ! grep -q "log_driver.*stdout" config/config.inc.php; then
+    echo "\$config['log_driver'] = 'stdout';" >> config/config.inc.php
+fi
+if ! grep -q "db_dsnw" config/config.inc.php; then
+    echo "\$config['db_dsnw'] = 'sqlite:////var/roundcube/sqlite.db?mode=0644';" >> config/config.inc.php
+fi
+
+# Run the steps necessary to actually use the repository code.
+
+# Install dependencies
+if [[ ! -f composer.json ]]; then
+    # For older versions of Roundcubemail.
+    cp -v composer.json-dist composer.json
+fi
+composer --prefer-dist --no-interaction --optimize-autoloader install
+
+# Download external Javascript dependencies.
+bin/install-jsdeps.sh
```

Comment:
Already done during build. Is this still necessary here?

Reply:
The dependencies might have changed since the image was built (or even between branches), so: Yes, it's needed.
Technically we could strip it from the Dockerfile, but since the dependencies in practice don't change often, it saves time to install them during the image build (and then starting the container doesn't require network access)